subprojects/maven/src/integTest/groovy/org/gradle/api/publish/maven/MavenPublishJavaRetriesIntegTest.groovy

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.publish.maven
+
+import org.gradle.integtests.fixtures.executer.ExecutionResult
+import org.gradle.integtests.fixtures.publish.maven.AbstractMavenPublishIntegTest
+import org.gradle.test.fixtures.keystore.TestKeyStore
+import org.gradle.test.fixtures.server.http.HttpServer
+import org.gradle.test.fixtures.server.http.MavenHttpModule
+import org.gradle.test.fixtures.server.http.MavenHttpRepository
+import org.junit.Rule
+
+class MavenPublishJavaRetriesIntegTest extends AbstractMavenPublishIntegTest {
+
+    TestKeyStore keyStore
+
+    @Rule public final HttpServer server = new HttpServer()
+
+    MavenHttpRepository mavenRemoteRepo
+    MavenHttpModule module
+
+    def setup() {
+        keyStore = TestKeyStore.init(file("keystore"))
+        server.start()
+
+        mavenRemoteRepo = new MavenHttpRepository(server, "/repo", mavenRepo)
+        module = mavenRemoteRepo.module("org.gradle.test", "testMavenRetries", "1.9")
+    }
+
+    def cleanup() {
+        server.stop()
+    }
+
+    def "should publish with intermittent network issues"() {
```

Comment:
The test needs to be annotated with `@ToBeFixedForInstantExecution`

Reply:
https://github.com/gradle/gradle/pull/12127/commits/633967df88c13765d5fe60a5cda33dc898a739cc